### PR TITLE
KM94 Add declarrative cluster setup

### PR DESCRIPTION
**Pull request by deifyed**
## Description

This PR implements the declarrative cluster approach as explained in [this](https://trello.com/c/k8LyFJLX/94-allow-for-programmatic-creation-of-clusters) task. 

By using `okctl scaffold cluster` you can now scaffold a cluster declaration which can be used to produce a cluster with the `okctl apply cluster` command. 

## Motivation and Context

The motivation is two-fold:
1. To lay groundwork for automated nightly testing by being able to create a cluster programmatically
2. Because CLI wizards are lame

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated the release notes (for the next release).

### CODE REVIEW

**Comment by deifyed** [comment]
Reserving update to documentation to when we can create the whole cluster this way, e.g.: after we implement automatic dns delegation / nameserver dance

File: cmd/okctl/applycluster.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+			}
+
+			outputDir, _ := o.GetRepoOutputDir(opts.Declaration.Metadata.Environment)
+
+			repoDir, err := o.GetRepoDir()
+			if err != nil {
+				return fmt.Errorf("could not get Repository dir: %w", err)
+			}
+
+			desiredGraph := controller.CreateDesiredStateGraph(opts.Declaration)
```
Comment (yngvark) [inline cmd/okctl/applycluster.go:108]: graph -> tree

Actually this goes everywhere.

File: pkg/controller/reconciler/reconsiler.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
+// Package reconciler contains different reconcilers for each of the necessary resources
```
Comment (yngvark) [inline pkg/controller/reconciler/reconsiler.go:1]: File spelling - reconsiler -> reconciler

**Review by yngvark — APPROVED** [review]
See comments, otherwise good (as this have been through PR before).

**Pull request merged**